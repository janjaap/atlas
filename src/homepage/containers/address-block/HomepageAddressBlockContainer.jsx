import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { fetchDataSelection } from '../../../header/ducks/search/search';

import HomepageAddressBlock from '../../components/address-block/HomepageAddressBlock';

const mapDispatchToProps = (dispatch) => bindActionCreators({
  onfetchDataSelection: fetchDataSelection
}, dispatch);

const HomepageAddressBlockContainer = (props) => (
  <HomepageAddressBlock
    onLinkClick={props.onfetchDataSelection}
  />
);

HomepageAddressBlockContainer.contextTypes = {
  store: PropTypes.object.isRequired
};

HomepageAddressBlockContainer.propTypes = {
  onfetchDataSelection: PropTypes.func.isRequired
};

export default connect(null, mapDispatchToProps)(HomepageAddressBlockContainer);
