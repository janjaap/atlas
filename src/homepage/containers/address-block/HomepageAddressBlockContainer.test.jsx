import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import HomepageAddressBlockContainer from './HomepageAddressBlockContainer';

describe('HomepageAddressBlockContainer', () => {
  it('should render', () => {
    const store = configureMockStore()({});
    const wrapper = shallow(<HomepageAddressBlockContainer />, { context: { store } }).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
