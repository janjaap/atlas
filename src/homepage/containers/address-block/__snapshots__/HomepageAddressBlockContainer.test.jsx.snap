// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageAddressBlockContainer should render 1`] = `
<HomepageAddressBlock
  onLinkClick={[Function]}
/>
`;
